ArjunaCore/arjuna/tests/classes/com/hp/mwtests/ts/arjuna/reaper/ReaperTerminateTestCase.java

```diff
@@ -0,0 +1,72 @@
+package com.hp.mwtests.ts.arjuna.reaper;
+
+import com.arjuna.ats.arjuna.common.Uid;
+import com.arjuna.ats.arjuna.coordinator.TransactionReaper;
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+public class ReaperTerminateTestCase extends ReaperTestCaseControl {
+
+    @Test
+    public void testReaperWait() throws Exception
+    {
+        runTest(true, 5);
+    }
+
+    @Test
+    public void testReaperForce() throws Exception
+    {
+        runTest(false, 5);
+    }
+
+    private void runTest(boolean waitForTransactions, int timeoutInSeconds) throws Exception
+    {
+        TransactionReaper reaper = TransactionReaper.transactionReaper();
+
+        // give the reaper worker time to start too
+
+        Thread.sleep(1000);
+
+        // create test reapable some of which will not respond immediately to cancel requests
+
+        Uid uid0 = new Uid();
+
+        // reapable0 will return CANCELLED from cancel and will rendezvous inside the cancel call
+        // so we can delay it. prevent_commit should not get called so we don't care about the arguments
+        ReaperTestCaseControl.TestReapable reapable0 = new ReaperTestCaseControl.TestReapable(uid0, true, false, false, false);
+        // reapable0 will return CANCELLED from cancel and will rendezvous inside the cancel call
```

Comment:
@jmfinelli I am not sure but this comment looks like a duplicate of 36/38? It looks like it is configured the same way though so maybe it will just be to change reapable0 to reapable1 in the comment on 39?

Reply:
Copy and paste errors :face_in_clouds: Thanks for spotting it

Reply:
Ah wait, I think this test should be removed entirely. Sorry, my bad